src/java/org/apache/cassandra/io/sstable/format/TOCComponent.java

```diff
@@ -78,16 +83,41 @@ public static Set<Component> loadTOC(Descriptor descriptor, boolean skipMissing)
     }
 
     /**
-     * Appends new component names to the TOC component.
+     * Updates the TOC file by reading existing component entries, merging them with the given components,
+     * sorting the combined list in lexicographic order for deterministic output.
+     *
+     * @param descriptor the SSTable descriptor for which to update the TOC
+     * @param components new components to merge into the TOC (existing TOC entries are preserved)
+     * @throws FSReadError if an I/O error occurs when reading the existing TOC file
+     * @throws FSWriteError if an I/O error occurs when creating or overwriting the TOC file
      */
-    public static void appendTOC(Descriptor descriptor, Collection<Component> components)
+    public static void updateTOC(Descriptor descriptor, Collection<Component> components)
     {
         File tocFile = descriptor.fileFor(Components.TOC);
-        try (FileOutputStreamPlus out = tocFile.newOutputStream(APPEND);
-             PrintWriter w = new PrintWriter(out))
+
+        // read existing entries if any
+        Path tocPath = tocFile.toPath();
+        Set<String> componentNames = new HashSet<>();
+        try
+        {
+            if (Files.exists(tocPath))
```

Comment:
Not necessary

````
Set<String> componentNames = new TreeSet<>(FileUtils.readLines(tocFile));
````

TreeSet will order it already. FileUtils.readLines will return empty list when not possible to read.